_kbarticles/2024-12-09-apt-buildpack-cache-issue.md

```diff
@@ -0,0 +1,30 @@
+---
+layout: post
+title: "Apt buildpack cache issue"
+date: December 9, 2024
+excerpt: Apt buildpack cache issue that can affect application deployment
+---
+
+## What does the buildpack cache do?
+
+The buildpack cache (aka "app cache," or "build artifacts cache") is a per-application cache that is stored in the blob (binary large object) store and recovered when an existing application is restaged (regardless of application code or buildpack changes).
+
+Different buildpacks use this cache for different things. System buildpacks are cached on the cell hosting the application container and bind mounted read-only into each staging container. Buildpack URLs (to a git repo or a zip file) are downloaded every time application staging occurs.
+
+## The potential issue with the apt buildpack
+
+The potential issue involves the / directory disk allocation. The apt buildpack uses /tmp like other buildpacks. The / directory is allocated approximately 9% of the disk allocation. For most buildpacks, this is sufficient as all the possible packages that they could be installing is known with the only variable being the application code. The apt buildpack however, can install a vast number of packages including large packages for example the AWS CLI. When this occurs this allotted 9% of disk space is not sufficient.
```

Comment:
```suggestion
The potential issue involves the `/` directory disk allocation. The apt buildpack uses `/tmp` like other buildpacks. The `/` directory is allocated approximately 9% of the disk allocation. For most buildpacks, this is sufficient as all the possible packages that they could be installing is known with the only variable being the application code. The apt buildpack however, can install a vast number of large packages, suchas  the AWS CLI. Depending on how many packages are installed and their individual size, the allotted 9% of disk space may not be sufficient.
```

As written, I don't know if I understand. Is the apt buildpack using the `/` or `/tmp` directory? The text suggests that `/tmp` is used but then describes a problem with the `/` directory. Is it because `/tmp` is a child directory of `/`?

Reply:
Yes, it is because /tmp is a child directory of /.